app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/connection/ConnectionActionHandler.java

```diff
@@ -101,16 +101,14 @@ public Response enrichWithMetadata(
             for (Entry<String, Object> entry : props.entrySet()) {
                 if (entry.getValue() == null) {
                     properties.put(entry.getKey(), null);
-                } else if (entry.getValue() instanceof String) {
-                    properties.put(entry.getKey(), (String) entry.getValue());
                 } else if (entry.getValue() instanceof String[]) {
                     String value = StringUtils.join((String[]) entry.getValue(), ConfigurationProperty.MULTIPLE_SEPARATOR);
                     properties.put(entry.getKey(), value);
                 } else if (entry.getValue() instanceof Iterable) {
                     String value = StringUtils.join((Iterable) entry.getValue(), ConfigurationProperty.MULTIPLE_SEPARATOR);
                     properties.put(entry.getKey(), value);
                 } else {
-                    throw new IllegalArgumentException("Not supported property type " + entry.getValue().getClass());
+                    properties.put(entry.getKey(), entry.getValue().toString());
```

Comment:
is this test still relevant https://github.com/syndesisio/syndesis/blob/master/app/server/endpoint/src/test/java/io/syndesis/server/endpoint/v1/handler/connection/ConnectionActionHandlerTest.java#L305 if there is no more an IllegalArgumentException thrown?

Reply:
the test is effectively failing https://circleci.com/gh/syndesisio/syndesis/56107

Reply:
Corrected it a second ago :)